### PR TITLE
fix: PERCEPTION-1980: Use a centralized lock for all realsense operat…

### PR DESCRIPTION
**Pull request by DrCodeWizzard**
…ions

**Please ensure the above *guidelines for contributing* are met.**

Fixes Issue: #

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

**Review by zachfang — APPROVED** [review]

File: realsense2_camera/include/realsense_node_factory.h [hunk 1]
```diff
@@ -63,20 +65,21 @@ namespace realsense2_camera
...
     private:
         virtual void onInit() override;
         void tryGetLogSeverity(rs2_log_severity& severity) const;
         std::string parseUsbPortId(std::string usb_path) const;
         bool deviceMatches(rs2::device& dev, std::string& usb_port);
         void addDevice(rs2::device dev);
-        void removeDevice(const rs2::event_information& info);
         void resetAndShutdown();
-
+        int device_iteration = 0;
```
Comment (dantwinkler) [inline realsense2_camera/include/realsense_node_factory.h]: Should prefer uint32_t in general.

File: realsense2_camera/src/base_realsense_node.cpp [hunk 2]
```diff
@@ -483,6 +499,29 @@ void BaseRealSenseNode::publishAlignedDepthToOthers(rs2::frame depth_frame, cons
 }
 
 void BaseRealSenseNode::setupStreams()
+{
+
```
Comment (dantwinkler) [inline realsense2_camera/src/base_realsense_node.cpp]: Should remove extra newlines.

File: realsense2_camera/src/base_realsense_node.cpp [hunk 3]
```diff
@@ -483,6 +499,29 @@ void BaseRealSenseNode::publishAlignedDepthToOthers(rs2::frame depth_frame, cons
...
+    ROS_INFO("setupStreams... NOT IMPLEMENTED");
+
+}
+
+void BaseRealSenseNode::stopTopics() {
+    ROS_INFO("shutting down streams");
+    for(auto&& sens : _sensors) {
```
Comment (dantwinkler) [inline realsense2_camera/src/base_realsense_node.cpp]: Should have space around ()

File: realsense2_camera/src/base_realsense_node.cpp [hunk 4]
```diff
@@ -483,6 +499,29 @@ void BaseRealSenseNode::publishAlignedDepthToOthers(rs2::frame depth_frame, cons
...
+        if (_enable[sens.first]) {
+        try {
```
Comment (dantwinkler) [inline realsense2_camera/src/base_realsense_node.cpp]: { should be on new lines.

**Review by zachfang — APPROVED** [review]

**Pull request merged**